UnityProject/Storybook/Assets/Code/Scripts/Game/Player/PlayerMover.cs

```diff
@@ -0,0 +1,338 @@
+            }
+
+            // Only allow the player to select a door if the door is enabled for the current room 
+            if (isWestDoorEnabled && GUILayout.Button("Move Left"))
+            {
+                int newRoomLocX = m_currentRoomLoc.X;
+                int newRoomLocY = m_currentRoomLoc.Y - 1;
+                photonView.RPC("CreateRoomOnMaster", PhotonTargets.MasterClient, newRoomLocX, newRoomLocY, RoomObject.DoorIndex.WestDoor);
+            }
+
+            // Only allow the player to select a door if the door is enabled for the current room 
+            if (isEastDoorEnabled && GUILayout.Button("Move Right"))
+            {
+                int newRoomLocX = m_currentRoomLoc.X;
+                int newRoomLocY = m_currentRoomLoc.Y + 1;
+                photonView.RPC("CreateRoomOnMaster", PhotonTargets.MasterClient, newRoomLocX, newRoomLocY, RoomObject.DoorIndex.EastDoor);
+            }
+
+            // Only allow the player to select a door if the door is enabled for the current room 
+            if (isSouthDoorEnabled && GUILayout.Button("Move Down"))
+            {
+                int newRoomLocX = m_currentRoomLoc.X + 1;
+                int newRoomLocY = m_currentRoomLoc.Y;
+                photonView.RPC("CreateRoomOnMaster", PhotonTargets.MasterClient, newRoomLocX, newRoomLocY, RoomObject.DoorIndex.SouthDoor);
+            }
+        }
+    }
+
+    /// <summary>
+    /// The list of players that are currently in the dungeon
+    /// </summary>
+    public List<NetworkNodeMover> WorldPlayers
+    {
+        get { return m_worldPlayers; }
+        set { m_worldPlayers = value; }
+    }
+
+    /// <summary>
+    /// Sets the target nodes of the all the players to the corresponding player nodes
+    /// </summary>
+    private void _SetTargetNodesForPlayers()
+    {
+        if (m_worldPlayers.Count >= 1)
+        {
+            m_worldPlayers[0].TargetNode = m_currentRoom.Player1Node;
+            m_worldPlayers[0].IsAtTarget = false;
+        }
+        if (m_worldPlayers.Count >= 2)
+        {
+            m_worldPlayers[1].TargetNode = m_currentRoom.Player2Node;
+            m_worldPlayers[1].IsAtTarget = false;
+        }
+        if (m_worldPlayers.Count >= 3)
+        {
+            m_worldPlayers[2].TargetNode = m_currentRoom.Player3Node;
+            m_worldPlayers[2].IsAtTarget = false;
+        }
+        if (m_worldPlayers.Count >= 4)
+        {
+            m_worldPlayers[3].TargetNode = m_currentRoom.Player4Node;
+            m_worldPlayers[3].IsAtTarget = false;
+        }
+    }
+
+    /// <summary>
+    /// Moves the camera to the new position and rotation
+    /// Called when the players switch rooms
+    /// </summary>
+    /// <param name="newCamPos">The new position for the camera</param>
+    /// <param name="newCamRot">The new rotation for the camera</param>
+    [PunRPC]
+    public void SwitchCameraRoom(Vector3 newCamPos, Quaternion newCamRot)
```

Comment:
The camera can just be parented to the player mover (or we can write a script on the camera to have it follow a target or whatever) don't manage it here.